SafeguardSessions/Scripts/Bootstrap.ps1

```diff
@@ -50,12 +50,33 @@ try {
 
     ConvertTo-Json $config -Depth 1 | Set-Content $VENV_DIR\config.json
 
-    Write-Output "Installing .NET SDK"
+    Write-Output "Installing .NET SDK as a background job"
     if (-Not (Test-Path($(Get-Alias -Name dotnet-install).Definition))) {
         Invoke-WebRequest -Uri https://dotnet.microsoft.com/download/dotnet/scripts/v1/dotnet-install.ps1 -OutFile $(Get-Alias -Name dotnet-install).Definition
     }
 
-    & dotnet-install -InstallDir $DOTNET -NoCdn -NoPath -Verbose -Version $packageVersions.DotnetVersion
+    $job = Start-Job -Name DotnetInstall -ScriptBlock {
+        switch ($input) {
+            { $true } {
+                & $_.DotnetInstall -InstallDir $_.InstallDir -NoCdn -NoPath -Verbose -Version $_.Version
+            }
+        }
+    } -InputObject @{
+        DotnetInstall = $(Get-Alias -Name dotnet-install).Definition
+        InstallDir    = $DOTNET
+        Version       = $packageVersions.DotnetVersion
+    }
+
+    $JobIDs = $(Get-Job -Name DotnetInstall -IncludeChildJob).Id
+
+    Wait-Job -Id $JobIDs -Timeout 600
+    Receive-Job -Id $JobIDs -ErrorAction Stop
+
+    if ($(Get-Job -Id $JobIDs).Where({ $_.State -eq "Running" }, "First").Count -gt 0) {
+        Write-Host "Timeout reached. Stopping the dotnet installer download job."
+        Stop-Job -Id $JobIDs
+        throw "Download aborted. Timeout reached."

```

Comment:
I would be more specific, e.g.: Failed to download and install the .NET SDK within 600 seconds.

Reply:
Done!